bitcoin/src/light/wallet.rs

```diff
@@ -57,7 +59,10 @@ impl GetSerializeSize for Witness {
 fn dummy_sign_input(txin: &mut TxIn, public_key: PublicKey) {
     // create a dummy signature that is a valid DER-encoding
     let dummy_signature = {
-        let m_r_len = 32;
+        // it is possible to "grind the signature" to encode an r-value
+        // in 32 bytes, but to be safe we use the default of 33 bytes
+        // which means that we may overestimate fees
+        let m_r_len = 33;
```

Comment:
This is where Bitcoin Core determines that length: https://github.com/bitcoin/bitcoin/blob/40e1c4d4024b8ad35f2511b2e10bf80c5531dfde/src/wallet/wallet.cpp#L1653

Reply:
There is some complex logic to determine whether an input transaction is "external" but this is difficult replicate so I opted to always just uses the maximum.